omgeo/services/__init__.py

```diff
@@ -920,3 +920,87 @@ def _geocode(self, pq):
             c.geoservice = self.__class__.__name__
             returned_candidates.append(c)
         return returned_candidates
+
+class Mapzen(GeocodeService):
+    """
+    Class to geocode using `Mapzen search service
+    <https://mapzen.com/projects/search>`_.
+
+    Settings used by the Mapzen GeocodeService object include:
+     * api_key --  The API key used to access search service.
+
+    """
+    _wkid = 4326
+
+    DEFAULT_PREPROCESSORS = [ReplaceRangeWithNumber()] # 766-68 Any St. -> 766 Any St.
+
+    def __init__(self, preprocessors=None, postprocessors=None, settings=None):
+        if settings.has_key('api_version'):
+            self._api_version = 'v' + str(settings['api_version'])
+        else:
+            self._api_version = 'v1'
+
+        if settings.has_key('instance_url'):
+            self._base_url = settings['instance_url']
+        else:
+            self._base_url = 'https://search.mapzen.com'
+
+        self._default_endpoint = self._base_url + '/' + \
```

Comment:
Since there are multiple parts to the URL how do you feel about something like this:

```python
        self._default_endpoint = urljoin(self._base_url,
                                    os.path.join(self._api_version, 'search'))
        self._key_endpoint = urljoin(self._base_url,
                                os.path.join(self._api_version, 'place'))
        self._endpoint = self._default_endpoint
```